File: 1557-check-if-a-string-contains-all-binary-codes-of-size-k/check-if-a-string-contains-all-binary-codes-of-size-k.cpp
class Solution {
public:
    bool hasAllCodes(string s, int k) {
        int n = s.length();
        int count = pow(2,k);
        unordered_set<string>st;
        for(int i=k;i<=n;i++){
            string sb = s.substr(i-k,k);
            if(st.count(sb) == false){
                st.insert(sb);
                count--;
            }
            if(count == 0){
                return true;
            }
        }
        return false;
    }
};

File: 1557-check-if-a-string-contains-all-binary-codes-of-size-k/check-if-a-string-contains-all-binary-codes-of-size-k.py
class Solution:
    def hasAllCodes(self, s: str, k: int) -> bool:
        st : set[str] = set()
        count = pow(2 , k)
        n = len(s)
        for i in range(k,n+1,1):
            sb = s[i-k : i]
            if(sb not in st):
                st.add(sb)
                count -= 1
            if(count == 0):
                return True
        return False